packages/zoltan2/core/src/input/Zoltan2_BasicIdentifierAdapter.hpp

```diff
@@ -116,31 +124,112 @@ template <typename User>
   BasicIdentifierAdapter(lno_t numIds, const gno_t *idPtr):
       numIds_(numIds), idList_(idPtr), weights_() {}
 
+  /*! \brief Constructor
+   *  \param ids should point to a View of identifiers.
+   *  \param weights  a list of pointers to arrays of weights.
+   *      The number of weights per identifier is assumed to be
+   *      \c weights.extent(0).
+   *  \param weightStrides  a list of strides for the \c weights.
+   *     The weight for weight index \c n for \c ids[k] should be
+   *     found at <tt>weights[n][weightStrides[n] * k]</tt>.
+   *     If \c weightStrides.size() is zero, it is assumed all strides are one.
+   *
+   *  The values pointed to the arguments must remain valid for the
+   *  lifetime of this Adapter.
+   */
+  BasicIdentifierAdapter(
+    Kokkos::View<gno_t*, device_t> &ids,
+    Kokkos::View<scalar_t**, device_t> &weights);
+
   ////////////////////////////////////////////////////////////////
   // The Adapter interface.
   ////////////////////////////////////////////////////////////////
 
-  size_t getLocalNumIDs() const { return numIds_; }
+  size_t getLocalNumIDs() const {
+    if (using_kokkos) {
+      return idsView_.extent(0);
+    } else {
+      return numIds_;
+    }
+  }
 
-  void getIDsView(const gno_t *&Ids) const { Ids = idList_; }
+  void getIDsView(const gno_t *&ids) const {
+    if (using_kokkos) {
+      auto h_ids = Kokkos::create_mirror_view(idsView_);
+      Kokkos::deep_copy(h_ids, idsView_);
+      ids = h_ids.data();
```

Comment:
Returns dangling pointer to local variable. That's an additional point in favor for ditching `using_kokkos` var

Reply:
Done